plugins/parsers/prometheusremotewrite/parser.go

```diff
@@ -0,0 +1,101 @@
+package prometheusremotewrite
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/common/model"
+)
+
+type Parser struct {
+	DefaultTags map[string]string
+}
+
+func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
+	var metrics []telegraf.Metric
+	var err error
+	var req prompb.WriteRequest
+
+
+	if err := proto.Unmarshal(buf, &req); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal request body: %s", err)
+	}
+
+
+	now := time.Now()
+
+	for _, ts := range req.Timeseries {
+		tags := map[string]string{}
+		for key, value := range p.DefaultTags {
+			tags[key] = value
+		}
+
+		for _, l := range ts.Labels {
+			tags[l.Name] = l.Value
+		}
+
+		metricName := tags[model.MetricNameLabel]
+		delete(tags, model.MetricNameLabel)
+
+		for _, s := range ts.Samples {
```

Comment:
As we approximately know how many metrics we expect we might want to preallocate the array with the number of samples, i.e. `capacity = len(ts.Samples)`  and `len = 0`.